### ISSUE TITLE
ErrorException Undefined array key "REQUEST_METHOD" with Laravel Octane with Swoole

### ISSUE THREAD

**Issue by gabrielboliveira** [inspector-apm/inspector-laravel#45]
**Describe the bug**

I'm testing Inspector.dev as part of a Laravel Octane + Swoole POC. I've installed in a somewhat fresh installation of Laravel, using Swoole as runtime with Laravel Octane.

We've followed [this article](https://inspector.dev/newrelic-alternative-for-monitoring-laravel-octane/) from Inspector.dev blog, but all requests fails with the following Exception:

```
Undefined array key "REQUEST_METHOD" {"exception":"[object] (ErrorException(code: 0): Undefined array key \"REQUEST_METHOD\" at /app/vendor/inspector-apm/inspector-php/src/Models/Partials/Request.php:16)
```

It looks like Swoole doesn't have all `$_SERVER` keys, see https://github.com/zendframework/zend-expressive-swoole/issues/33.

**To Reproduce**

It is kinda hard to replicate since it is specific for Swoole, but a fresh Laravel installation with Swoole as runtime could replicate the issue.

1. Run Laravel with Octane, PHP 8.3 and Swoole as runtime.
2. Perform any request
3. See the Exception

**Expected behavior**
The `inspector-apm/inspector-php` should properly receive the request method, even if `$_SERVER` doesn't have that key.

The `Models/Partials/Request.php` from `inspector-apm/inspector-php` could receive the method, versions, cookies and etc from optional constructor attributes, or read from other places that are available.

**Screenshots**

<img width="918" alt="Captura de Tela 2024-07-10 às 09 53 29" src="https://github.com/inspector-apm/inspector-laravel/assets/11093090/3a3c0ea2-8c1b-4bca-835c-55c7e5d5239f">

**Web (please complete the following information):**
 - Browser: Any
 - Package Version: 
   - inspector-apm/inspector-laravel: 4.9.2
   - inspector-apm/inspector-php: 3.8.1
 - Laravel version: 11.11.0
 - PHP Version: 8.3.8
 - Swoole Version:  5.1.3

**Additional context**

This might be a `inspector-apm/inspector-php` bug instead of this specific package, however since this scenario was replicated on Laravel Octane, I decided to open on this repository.

**Comment by ilvalerione** [comment]
Hi @gabrielboliveira I reproduced your scenario but I didn't see any error. The Laravel installation works as expected and Http Requests are reported to Inspector with no errors.

**Comment by gabrielboliveira** [comment]
@ilvalerione That's weird. I've forgot to mention that we're running on Docker, but that shouldn't be an issue, right?

[This is the Dockerfile I'm using for testing](https://github.com/somosyampi/docker-php-fpm/blob/php-swoole/8.3-swoole-prod/Dockerfile). Then, using this command to start Octane:

```
php artisan octane:start --server=swoole --host=0.0.0.0 --port=80 --task-workers=6
```

Also, not running `nginx` as reverse proxy. Pointing directly to Swoole.

**Comment by ilvalerione** [comment]
Docker shouldn't be the problem. I see a potential issue in the absence of NGINX. We can add a fallback eventually, but in this case you will never see the request details like headers, cookies, URLs.

**Comment by ilvalerione** [comment]
What kind of REQUEST information are present in your php global variables when a request comes into your app? These variables are also used by Laravel to create the Request instance.

**Comment by gabrielboliveira** [comment]
I was able to fix it by extending the `InspectorOctaneMiddleware` and adding the missing and expected keys:

```php
<?php

namespace App\Http\Middleware;

use Closure;
use Inspector\Laravel\Middleware\InspectorOctaneMiddleware;

class InspectorMonitoringMiddleware extends InspectorOctaneMiddleware
{
    /**
     * Handle an incoming request.
     *
     * @param \Illuminate\Http\Request $request
     * @param \Closure $next
     * @return mixed
     * @throws \Exception
     */
    public function handle($request, Closure $next)
    {
        $_SERVER['REQUEST_METHOD'] = $request->getMethod();
        $_SERVER['REQUEST_URI'] = $request->getRequestUri();
        $_SERVER['SERVER_PROTOCOL'] = 'HTTP/' . $request->getProtocolVersion();
        $_SERVER['HTTP_HOST'] = $request->getHost();
        $_SERVER['REMOTE_ADDR'] = $request->getClientIp();
        $_SERVER['SERVER_NAME'] = $request->server('SERVER_NAME');
        $_SERVER['SERVER_PORT'] = $request->getPort();
        $_SERVER['QUERY_STRING'] = $request->getQueryString();
        $_SERVER['SCRIPT_NAME'] = $request->server('SCRIPT_NAME');

        return parent::handle($request, $next);
    }
}
```

I see a very similar behavior to what was described in [this issue](https://github.com/zendframework/zend-expressive-swoole/issues/33), where the `$_SERVER` request misses a good number of attributes.

**Comment by ilvalerione** [comment]
Thank you for sharing, it could be definitely a good resource for others that stumble into this borderline issue.

**Comment by andreinocenti** [comment]
I had the same issue reported by @gabrielboliveira. The difference is that we updated our laravel 10 app to laravel 11. 
We used the fix suggested by Gabriel and it worked perfectly.

@ilvalerione Maybe this issue should be reopened to be treated in a future fix

**Comment by ilvalerione** [comment]
From our side the problem was that it wasn't able to reproduce the issue. We tested it in a Laravel 11 Application + Octane (Swoole) and it worked as expected. Can you report here the exact version of PHP, Laravel, Octane?

**Comment by andreinocenti** [comment]
@ilvalerione 
It happened on both PHP 8.2.20 and 8.3.9
Laravel 11.19
Octane 5.0.3 (php 8.2.19) and 5.1.3 (php 8.3.9)

Something I just realized, at Inspector docs, it says to put the Laravel 11 config at the kernel.php. But Kernel.php is removed on laravel 11 (it still works if you keep the file). Now the config is done at bootstrap/app.php

my app.php (using Gabriel's solution):
```php
return Application::configure(basePath: dirname(__DIR__))
    ->withRouting(
        api: __DIR__ . '/../routes/api.php',
        commands: __DIR__ . '/../routes/console.php',
        health: '/up',
    )->withMiddleware(function (Middleware $middleware) {
        $middleware->appendToGroup('api', InspectorMonitoringMiddleware::class);
        $middleware->validateCsrfTokens(
            except: ['stripe/*']
        )->statefulApi();

    // Append the middleware
    })->withExceptions(function (Exceptions $exceptions) {
        //
    })->create();
```

**Comment by ilvalerione** [comment]
I will report the result ofter testing with these versions.

**Comment by tomykho** [comment]
Same issue happened with Laravel Vapor + Octane.
This fix worked https://github.com/inspector-apm/inspector-laravel/issues/45#issuecomment-2225460697, but I am not sure modifying global variable like $_SERVER is a good idea.

**Comment by martin-juul** [comment]
This is still an issue with laravel 11.26 and swoole 5.1.4

**Comment by ilvalerione** [comment]
I released a patch (4.9.12) that incorporate this solution natively into the `InspectorOctaneMiddleware`
